gems/smithy-schema/lib/smithy-schema/document.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+require_relative 'document_utils'
+
+module Smithy
+  module Schema
+    # A Smithy document type, representing typed or untyped data from Smithy data model.
+    # ## Document types
```

Comment:
I could add more details (e.g. code examples) but I might punt that to the Smithy-Ruby Wiki instead.